src/components/upload/nz-basic-upload.component.ts

```diff
@@ -0,0 +1,150 @@
+import { Component, EventEmitter, Input, Output, OnInit, ElementRef, ViewChild, ViewEncapsulation } from '@angular/core';
+import { UidService } from './uid/uid.service';
+import defaultRequest from './request';
+import { attrAccept } from './utils';
+
+@Component({
+    selector: 'nz-basic-upload',
+    providers: [UidService],
+    templateUrl: './nz-basic-upload.component.html',
+    styleUrls: [
+        './style/index.less'
+    ],
+    encapsulation: ViewEncapsulation.None
+})
+export class NzBasicUploadComponent implements OnInit {
+
+    _classMap;
+    _prefixCls = 'ant-upload';
+    _reqs = {};
+
+    @Input() accept: string;
+    @Input() action: string;
+    @Input() autoUpload = true;
+    @Input() beforeUpload: (file: File, files: FileList) => boolean | Promise<any> = null;
+    @Input() customRequest: any;
+    @Input() data: any;
+    @Input() disabled = false;
+    @Input() headers: string | { [name: string]: string | string[] } = {};
+    @Input() multiple = false;
+    @Input() name: string;
+    @Input() withCredentials = false;
+
+    @Output() onError = new EventEmitter<any>();
+    @Output() onProgress = new EventEmitter<any>();
+    @Output() onRemove = new EventEmitter<any>();
+    @Output() onStart = new EventEmitter<any>();
+    @Output() onSuccess = new EventEmitter<any>();
+
+    @ViewChild('input') inputElement: ElementRef;
+
+
+
+    constructor(private uidService: UidService) { }
+
+    ngOnInit() {
+        this.setClassMap();
+    }
+
+    setClassMap(): void {
+        this._classMap = {
+            [`${this._prefixCls}`]: true,
+            [`${this._prefixCls}-disabled`]: this.disabled,
+        };
+    }
+
+    onChange = (e) => {
+        const files: FileList = e.target.files;
+        if (!files) { return };
+        this.uploadFiles(files);
+    }
+
+    uploadFiles(files: FileList) {
+        const postFiles = Array.prototype.slice.call(files);
+
+        postFiles.forEach((file) => {
+            file.uid = this.uidService.getUid();
+            this.onStart.emit(file);
+            if (this.autoUpload) {
+                this.upload(file, postFiles);
+            }
+        });
+    }
+
+    upload(file, fileList) {
+        if (!this.beforeUpload) {
+            this.post(file);
+        } else {
+            const before = this.beforeUpload(file, fileList);
+            if (before instanceof Promise && before && before.then) {
```

Comment:
Since you have judgement of `before.then` for all Promise like object, i think here is no necessary of `before instanceof Promise`

Reply:
Thanks, `nz-basic-upload.component.ts` won't be exposed, if not necessary. :smile:

Reply:
Thanks,I will correct them.

Reply:
@wilsoncook I try to remove `before instanceof Promise`.There is nothing wrong when it run. But TSLint throw
> [ts]
Property 'then' does not exist on type 'true | Promise<any>'.
  Property 'then' does not exist on type 'true'.

Reply:
@wilsoncook Locale has been done

Reply:
- [x] nz prefix